### PR TITLE
Integration test runs - client fixes

### PR DESCRIPTION
**Pull request by oleorhagen**
All miscellaneous changes, and nitpicks, and fixes.

### CODE REVIEW

File: mender-update/deployments/deployments.cpp [hunk 1]
```diff
@@ -285,12 +285,18 @@ error::Error DeploymentClient::PushStatus(
...
-			if (!ex_len) {
+			if (!content_length) {
 				log::Error("Failed to get content length from the status API response headers");
 				body_writer->SetUnlimited(true);
 			} else {
-				received_body->resize(ex_len.value());
+				auto ex_len = common::StringToLongLong(content_length.value());
+				if (!ex_len) {
+					log::Error(
+						"Failed to convert the content length from the status API response headers to an integer");
```
Comment (lluiscampos) [inline mender-update/deployments/deployments.cpp:295]: I would append to the log message `ex_len.error().String()`

Comment (lluiscampos) [reply mender-update/deployments/deployments.cpp:295]: Pre-existing,

Shouldn't this (and few lines above) be a warning instead? Setting the body_writer to be unlimited makes it in effect to recover.

File: mender-update/deployments/deployments.cpp [hunk 2]
```diff
@@ -449,12 +449,18 @@ error::Error DeploymentClient::PushLogs(
...
+			if (!content_length) {
+				log::Error("Failed to get content length from the status API response headers");
 				body_writer->SetUnlimited(true);
 			} else {
-				received_body->resize(ex_len.value());
+				auto ex_len = common::StringToLongLong(content_length.value());
+				if (!ex_len) {
+					log::Error(
+						"Failed to convert the content length from the status API response headers to an integer");
+					body_writer->SetUnlimited(true);
```
Comment (lluiscampos) [inline mender-update/deployments/deployments.cpp:460]: Same suggestion(s) than before :+1:

**Review by lluiscampos — APPROVED** [review]
Nice fixes! I am curious to hear how many tests you managed to ~run~ pass :smiley:

**Comment by oleorhagen** [comment]
> Nice fixes! I am curious to hear how many tests you managed to ~run~ pass 😃

Straight OOTB, none so far unfortunately, I've been cheating a bit to get them to run, like the certs don't work yet, but that will be todays fix, then we will see :)

Locally, I can do simple file updates with `demo` though :tada:

File: mender-update/deployments/deployments.cpp (hunk 1 above)
Comment (oleorhagen) [reply mender-update/deployments/deployments.cpp:295]: Not against, it certainly. I'll add a separate commit for it though

**Pull request merged**